src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,84 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.InvalidAgeException;
+import core.basesyntax.exception.LoginValidateException;
+import core.basesyntax.exception.PasswordValidateException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "user_login";
+    private static final String VALID_PASSWORD = "user_password";
+    private static final int VALID_AGE = 18;
+    private static final String INVALID_LOGIN = "log";
+    private static final String INVALID_PASSWORD = "1234";
+    private static final String INVALID_LOGIN_WITH_NO_FIRST_LETTER = "1invalidLogin";
+    private static final int INVALID_AGE = 17;
+
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+    private User user;
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setAge(VALID_AGE);
+        user.setLogin(VALID_LOGIN);
+        user.setPassword(VALID_PASSWORD);
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_nullUser_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> registrationService.register(null));
```

Comment:
This test is testing that our service is written poor 😅 
When we throw or expect an NPE - it is wrong, in every way possible, what we should do is add a validation to the service and check if user is null - throw our custom exception with a message that we can not accept the null user and thus he is not valid. Here we should assert this exact custom exception and it's message